pulser/sequence.py

```diff
@@ -12,32 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
 from collections import namedtuple
-from collections.abc import Iterable
+from collections.abc import Callable, Generator, Iterable, Set
 import copy
 from functools import wraps
 from itertools import chain
 import json
+from typing import Any, cast, NamedTuple, Optional, Tuple, Union
+from typing_extensions import Literal
+
 import warnings
 
 import numpy as np
+from numpy.typing import ArrayLike
 
 import pulser
+from pulser._seq_drawer import draw_sequence
+from pulser.channels import Channel
 from pulser.pulse import Pulse
 from pulser.devices import MockDevice
 from pulser.devices._device_datacls import Device
 from pulser.json.coders import PulserEncoder, PulserDecoder
 from pulser.json.utils import obj_to_dict
 from pulser.parametrized import Parametrized, Variable
-from pulser._seq_drawer import draw_sequence
+from pulser.register import Register
+
+QubitId = Union[int, str]
+PROTOCOLS = Literal['min-delay', 'no-delay', 'wait-for-all']
+
+
+class _TimeSlot(NamedTuple):
+    # Auxiliary class to store the information in the schedule
```

Comment:
You can change this to a docstring: 
```suggestion
    """Auxiliary class to store the information in the schedule."""
```